controllers/front/ReportComment.php

```diff
@@ -47,7 +47,9 @@ public function display()
             return false;
         }
 
-        $id_product_comment = (int) Tools::getValue('id_product_comment');
+        $content = trim(file_get_contents('php://input'));
```

Comment:
First, when replace jQuery Ajax by Fetch API in [list-comment.js](https://github.com/PrestaShop/productcomments/pull/189/files#diff-ee302def5e771b16c7c44e4a978fe52c51bf99a052510959e159648e23ef4373), we need replace `Tools::getValue` by `file_get_contents('php://input')` accordingly, because Fetch Post is sent as `application/json` not `application/x-www-form-urlencoded` nor `multipart/form-data`
Second, according to [from-Legacy-to-Future-architecture's](https://build.prestashop-project.org/news/2020/from-legacy-to-future-architecture-connecting-the-dots/) Homogenize the FO & BO architecture, `Tools` is an ancient legacy-based component, so this Tools removing just double the benefit of this PR. I really replace two legacy library usage by two native functions 😆

Reply:
Thank you for the explanations.

Mmmm 🤔 so that means that every module developer who wants to use Fetch will have to do the same. So we can expect a lot of `$content = trim(file_get_contents('php://input'));` that appears... until Symfony can be used on the front (because I'm quite sure Symfony Request object can handle this very well)

Reply:
In my opinion this is way overkill, what's the advantage of using fetch with a JSON body content?

Replacing `$.post` by fetch is a good call, but you can use `fetch` to perform a regular POST request with form vars instead of JSON body, it will simplify both the JS and PHP code

Reply:
I just follow this manual https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch . And I think we are in a Listing context not a Form context here.

Reply:
That's not the point, request with form vars are performed on many APIs regardless of what they are requesting, or the context of the request The `FormData` here is actually the JS native way to send POST parameters, nothing more, and POST variables are more convenient to use in PHP on the BO side

Requests based on JSON bodies are relevant for REST APIs for example, or in JS ecosystem because most framework already handle them natively. We are also using a JSON bdy in our new API under development, but because we are in a Symfony framework and also relying on APIPlatform which handles the parsing under the hood

But for native PHP in a legacy controller it's not really among the usual practices Most of the time simple API endpoints in PHP rely on `$_POST` variables, or at least all our current legacy controllers do

We have no easy tool to get a JSON body request on legacy controller, that's why you had to "hard code" the fetching and parsing based on `php://input` But I'm pretty sure the next developer who will see this won't even understand what's it used for 🤷‍♂️

So in my opinion, but I'm not the only one to decide, the PR should only focus on refactoring the JS side, as your PR title suggests, without impacting the server side controller So instead of sending a JSON body it should send POST variables and my suggestion simply shows how it's done in vanilla JS

Reply:
Alright. If you are so worried, we can switch this PR to draft, and continue to discuss more in the related discussion. Wdyt @matks  ?

Reply:
OK. Let's postpone this risky PR. I will find another way.